lib/kubernetes-deploy/runner.rb

```diff
@@ -393,22 +395,41 @@ def log_pruning(kubectl_output)
     end
 
     def confirm_context_exists
-      out, err, st = kubectl.run("config", "get-contexts", "-o", "name",
+      out, st, err = nil
+      3.times do
+        out, err, st = kubectl.run("config", "get-contexts", "-o", "name",
         use_namespace: false, use_context: false, log_failure: false)
+        break if st.success?
+        sleep 5
+      end
       available_contexts = out.split("\n")
       if !st.success?
         raise FatalDeploymentError, err
       elsif !available_contexts.include?(@context)
         raise FatalDeploymentError, "Context #{@context} is not available. Valid contexts: #{available_contexts}"
       end
+      confirm_context_valid_on_server
       @logger.info("Context #{@context} found")
     end
 
+    def confirm_context_valid_on_server
```

Comment:
@KnVerey Still working on cleaning this up but wanted to see if you had any reservations against this method?

Reply:
In general this is what I was thinking, yes. I think we should retry the call a few times if we get an error though. It also occurred to me that it could be less complex and just as helpful to use `log_failure: true` to get _all_ the failure messages in the output, and then just raise an error about what we were trying to do on the final failure. 

Other nit: "Context not found on server" isn't exactly accurate, because context is a client-side pairing of a user and a cluster. What we're detecting here is whether the cluster defined by the context in question is reachable. It doesn't terribly matter what command we use as long as we have a test that fails if the command we've chosen stops trying to hit the server.

Suggestion:

```ruby
def confirm_cluster_reachable
  success = false
  with_retries(3) do 
    _, _, st = kubectl.run("version", use_namespace: false, log_failure: true)
    success = st.success?
  end
  
  raise FatalDeploymentError, "Failed to reach server for #{@context}" unless success
end
```

Reply:
Updated as per your recommendation